packages/core/src/vivliostyle/css-cascade.ts

```diff
@@ -2009,6 +2009,127 @@ export class ContentPropVisitor extends Css.FilterVisitor {
     return new Css.Str(stringValue);
   }
 
+  /**
+   * CSS `leader()` function
+   * https://www.w3.org/TR/css-content-3/#leaders
+   */
+  visitFuncLeader(values: Css.Val[]): Css.Val {
+    let leader: string = "";
+    if (values[0] instanceof Css.Ident) {
+      switch (values[0].stringValue()) {
+        case "dotted":
+          leader = ".";
+          break;
+        case "solid":
+          leader = "_";
+          break;
+        case "space":
+          leader = " ";
+          break;
+      }
+    } else if (values[0] instanceof Css.Str) {
+      leader = values[0].stringValue();
+    }
+    if (leader.length == 0) {
+      return new Css.Str("");
+    }
+    if (!this.postLayoutLeaderRegistered) {
+      Plugin.registerHook(
+        Plugin.HOOKS.POST_LAYOUT_BLOCK,
+        this.postLayoutLeader.bind(this),
+      );
+      this.postLayoutLeaderRegistered = true;
+    }
+    return new Css.Expr(
+      new Exprs.Native(
+        null,
+        (): Exprs.Result => {
+          return leader.replace(" ", "\u00A0"); // nbsp
+        },
+        "viv-leader",
+      ),
+    );
+  }
+  postLayoutLeaderRegistered: boolean = false;
+  postLayoutLeader: Plugin.PostLayoutBlockHook = (
+    nodeContext: Vtree.NodeContext,
+    checkPoints: Vtree.NodeContext[],
+    column: Layout.Column,
+  ) => {
+    // we want to access the bottom block element, which contains single leader().
+    if (nodeContext.viewNode.nodeType !== 1) {
+      return;
+    }
+    const container = nodeContext.viewNode as Element;
+    const leaders = container.querySelectorAll(`span>span[viv-leader]`);
+    if (leaders.length != 1) {
+      return;
+    }
+    const e = leaders[0];
+    const pseudoAfter = e.parentElement;
+    const leader = e.getAttribute("viv-leader-value");
+    const previous = e.textContent;
+
+    // reset the expanded leader
+    e.textContent = leader;
+    pseudoAfter.style.paddingTop = "0";
+    pseudoAfter.style.paddingLeft = "0";
```

Comment:
paddingTopとpaddingLeftを"0"にしてるのは、縦書きとltr横書きのためかと思います。rtlのサポートも考慮すると、次のように論理プロパティを使うのがよいのでないでしょうか。

```suggestion
    pseudoAfter.style.paddingInlineStart = "0";
```